https://github.com/deedy5/ddgs

In [1]:
!pip install -U ddgs

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 40.3/40.3 kB 1.0 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 161.7/161.7 kB 4.7 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.3/3.3 MB 44.0 MB/s eta 0:00:00


In [2]:
!ddgs --help

Usage: ddgs [OPTIONS] COMMAND1 [ARGS]... [COMMAND2 [ARGS]...]...

  DDGS CLI tool.

Options:
  --help  Show this message and exit.

Commands:
  books    CLI function to perform a DDGS books metasearch.
  images   CLI function to perform a DDGS images metasearch.
  news     CLI function to perform a DDGS news metasearch.
  text     CLI function to perform a DDGS text metasearch.
  version  Print and return version.
  videos   CLI function to perform a DDGS videos metasearch.


https://github.com/deedy5/ddgs

In [3]:
from ddgs import DDGS

results = DDGS().text("python programming", max_results=5)
print(results)

[{'title': 'Python (programming language)', 'href': 'https://en.wikipedia.org/wiki/Python_(programming_language)', 'body': 'Python is a high-level, general-purpose programming language. Its design philosophy emphasizes code readability with the use of significant indentation. Python is dynamically type-checked and garbage-collected. It supports multiple programming paradigms, including structured (particularly procedural), object-oriented and functional programming.Guido van Rossum began working on Python in the late 1980s as a successor to the ABC programming language. Python 3.0, released in 2008, was a major revision and not completely backward-compatible with earlier versions. Beginning with Python 3.5, capabilities and keywords for typing were added to the language, allowing optional static typing. As of 2025, the Python Software Foundation supports Python 3.10, 3.11, 3.12, 3.13, and 3.14, following the projects annual release cycle and five-year support policy. Earlier versions i

In [ ]:
def text(
    query: str,
    region: str = "us-en",
    safesearch: str = "moderate",
    timelimit: str | None = None,
    max_results: int | None = 10,
    page: int = 1,
    backend: str = "auto",
) -> list[dict[str, str]]:
    """DDGS text metasearch.

    Args:
        query: text search query.
        region: us-en, uk-en, ru-ru, etc. Defaults to us-en.
        safesearch: on, moderate, off. Defaults to "moderate".
        timelimit: d, w, m, y. Defaults to None.
        max_results: maximum number of results. Defaults to 10.
        page: page of results. Defaults to 1.
        backend: A single or comma-delimited backends. Defaults to "auto".

    Returns:
        List of dictionaries with search results.
    """

In [4]:
from ddgs import DDGS
def text(
    query: str,
    region: str = "us-en",
    safesearch: str = "moderate",
    timelimit: str | None = None,
    max_results: int | None = 10,
    page: int = 1,
    backend: str = "auto",
) -> list[dict[str, str]]:
    """DDGS text metasearch.

    Args:
        query: text search query.
        region: us-en, uk-en, ru-ru, etc. Defaults to us-en.
        safesearch: on, moderate, off. Defaults to "moderate".
        timelimit: d, w, m, y. Defaults to None.
        max_results: maximum number of results. Defaults to 10.
        page: page of results. Defaults to 1.
        backend: A single or comma-delimited backends. Defaults to "auto".

    Returns:
        List of dictionaries with search results.
    """
results = DDGS().text("python programming", max_results=5)
print(results)

[{'title': 'Python (programming language)', 'href': 'https://en.wikipedia.org/wiki/Python_(programming_language)', 'body': 'Python is a high-level, general-purpose programming language. Its design philosophy emphasizes code readability with the use of significant indentation. Python is dynamically type-checked and garbage-collected. It supports multiple programming paradigms, including structured (particularly procedural), object-oriented and functional programming.Guido van Rossum began working on Python in the late 1980s as a successor to the ABC programming language. Python 3.0, released in 2008, was a major revision and not completely backward-compatible with earlier versions. Beginning with Python 3.5, capabilities and keywords for typing were added to the language, allowing optional static typing. As of 2025, the Python Software Foundation supports Python 3.10, 3.11, 3.12, 3.13, and 3.14, following the projects annual release cycle and five-year support policy. Earlier versions i



[{'title': 'Python (programming language)', 'href': 'https://en.wikipedia.org/wiki/Python_(programming_language)', 'body': 'Python is a high-level, general-purpose programming language. Its design philosophy emphasizes code readability with the use of significant indentation. Python is dynamically type-checked and garbage-collected. It supports multiple programming paradigms, including structured (particularly procedural), object-oriented and functional programming.Guido van Rossum began working on Python in the late 1980s as a successor to the ABC programming language. Python 3.0, released in 2008, was a major revision and not completely backward-compatible with earlier versions. Beginning with Python 3.5, capabilities and keywords for typing were added to the language, allowing optional static typing. As of 2025, the Python Software Foundation supports Python 3.10, 3.11, 3.12, 3.13, and 3.14, following the projects annual release cycle and five-year support policy. Earlier versions in the 3.x series have reached end-of-life and no longer receive security updates. Python has gained widespread use in the machine learning community. It is widely taught as an introductory programming language. Since 2003, Python has consistently ranked in the top ten of the most popular programming languages in the TIOBE Programming Community Index, which ranks based on searches in 24 platforms.'}, {'title': 'Python (programming language)', 'href': 'https://grokipedia.com/page/Python_(programming_language)', 'body': 'Python is an interpreted, high-level, general-purpose programming language designed for code readability and simplicity, featuring a dynamic type system and support for...'}, {'title': 'Programming for Everybody (Getting Started with Python) |', 'href': 'https://www.coursera.org/learn/python', 'body': '... aims to teach everyone the basics of programming computers using Python ... In this module you will set things up so you can write Python programs.'}, {'title': 'Python Programming - Wikibooks, open books for an open world', 'href': 'https://en.wikibooks.org/wiki/Python_Programming', 'body': 'A printable version of Python Programming is available. ... Retrieved from " https://en.wikibooks.org/w/index.php?title= Python _ Programming &oldid ...'}, {'title': 'Computer Programming for Everybody | Python.org', 'href': 'https://www.python.org/doc/essays/cp4e/', 'body': "Python 's programming environment and documentation are less than ideal for teaching to novices. ... we will initially use the Python programming ..."

In [5]:
from ddgs import DDGS
import json
import time

def text_search(
    query: str,
    region: str = "us-en",
    safesearch: str = "moderate",
    timelimit: str | None = None,
    max_results: int = 10,
    proxies: dict | None = None
) -> list[dict]:
    """
    دالة بحث محسنة مع معالجة الأخطاء وإدارة الاتصال.
    """
    try:
        # استخدام with لضمان إدارة الموارد بشكل صحيح
        with DDGS(proxies=proxies) as ddgs:
            # استدعاء دالة البحث
            results = ddgs.text(
                keywords=query,
                region=region,
                safesearch=safesearch,
                timelimit=timelimit,
                max_results=max_results
            )
            # تحويل المولد (generator) إلى قائمة
            return list(results)

    except Exception as e:
        print(f"حدث خطأ أثناء البحث عن '{query}': {e}")
        return []

# --- تجربة الكود ---

# 1. بحث عادي
print("--- جاري البحث ---")
results = text_search("python programming", max_results=5)

# 2. طباعة النتائج بشكل جميل (Pretty Print)
if results:
    print(json.dumps(results, indent=4, ensure_ascii=False))
else:
    print("لم يتم العثور على نتائج.")

# 3. معالجة النتائج (مثال: استخراج العناوين فقط)
print("\n--- العناوين المستخرجة ---")
for item in results:
    print(f"- {item.get('title')}")

--- جاري البحث ---
حدث خطأ أثناء البحث عن 'python programming': DDGS.__init__() got an unexpected keyword argument 'proxies'
لم يتم العثور على نتائج.

--- العناوين المستخرجة ---


In [ ]:
from ddgs import DDGS

results = DDGS().text("python programming", max_results=5)
print(results)

In [6]:
from ddgs import DDGS

In [7]:
def books(
    query: str,
    max_results: int | None = 10,
    page: int = 1,
    backend: str = "auto",
) -> list[dict[str, str]]:
    """DDGS books metasearch.

    Args:
        query: news search query.
        max_results: maximum number of results. Defaults to 10.
        page: page of results. Defaults to 1.
        backend: A single or comma-delimited backends. Defaults to "auto".

    Returns:
        List of dictionaries with news search results.
    """

In [8]:
results = DDGS().books(query="sea wolf jack london", page=1, backend="auto")
print(results)
[
    {
        'title': 'The Sea-Wolf',
        'author': 'Jack London',
        'publisher': 'DigiCat, 2022',
        'info': 'English [en], .epub, 🚀/zlib, 0.5MB, 📗 Book (unknown)',
        'url': 'https://annas-archive.li/md5/574f6556f1df6717de4044e36c7c2782',
        'thumbnail': 'https://s3proxy.cdn-zlib.sk//covers299/collections/userbooks/da4954486be7c2b2b9f70b2aa5bcf01292de3ea510b5656f892821950ded9ada.jpg',
    }, ...
]

[{'title': 'The Sea Wolf', 'author': 'Jack London [London, Jack]', 'publisher': 'Feedbooks, 1904', 'info': 'English [en] · EPUB · 0.8MB · 1904 · 📕 Book (fiction) · 🚀/lgli/zlib ·', 'url': 'https://annas-archive.li/md5/d54357cb938ae756bcb49686424390ca', 'thumbnail': 'https://s3proxy.cdn-zlib.sk//covers299/collections/userbooks/8a9153ea9f817df491bf4760251e762ca5c89a0b0b1d2b2efc1ab619809bea3d.png'}, {'title': 'The Sea Wolf', 'author': 'Jack London', 'publisher': 'The Macmillan Company, 1946', 'info': 'English [en] · PDF · 16.2MB · 1946 · 📗 Book (unknown) · 🚀/ia ·', 'url': 'https://annas-archive.li/md5/1fe5ae31b3eedd7c6ee4791528a9018a', 'thumbnail': 'https://archive.org/download/bwb_P9-EJB-266/__ia_thumb.jpg'}, {'title': 'The Sea-Wolf', 'author': 'Jack London', 'publisher': 'DigiCat, 2022', 'info': 'English [en] · EPUB · 0.5MB · 2022 · 📗 Book (unknown) · 🚀/zlib ·', 'url': 'https://annas-archive.li/md5/574f6556f1df6717de4044e36c7c2782', 'thumbnail': 'https://s3proxy.cdn-zlib.sk//covers299/co

[{'title': 'The Sea-Wolf',
  'author': 'Jack London',
  'publisher': 'DigiCat, 2022',
  'info': 'English [en], .epub, 🚀/zlib, 0.5MB, 📗 Book (unknown)',
  'url': 'https://annas-archive.li/md5/574f6556f1df6717de4044e36c7c2782',
  'thumbnail': 'https://s3proxy.cdn-zlib.sk//covers299/collections/userbooks/da4954486be7c2b2b9f70b2aa5bcf01292de3ea510b5656f892821950ded9ada.jpg'},
 Ellipsis]

In [ ]:
from ddgs import DDGS

results = DDGS().text("python programming", max_results=5)
print(results)

In [9]:
from ddgs import DDGS

# إنشاء كائن DDGS
ddgs = DDGS()

# البحث عن صور "أطفال"
results = ddgs.images(
    query="أطفال",
    region="us-en",        # يمكنك تغييرها حسب بلدك، مثلاً "xa-ar" للعربية
    safesearch="moderate", # أو "on" لفلترة أكثر صرامة
    max_results=10
)

# طباعة النتائج
for img in results:
    print(img["title"], img["image"])

إنتاج أغنية بصوت أطفال للحضانة أو للتعليم أو للمسرحيات - خمسات https://khamsat.hsoubcdn.com/images/services/2848438/e9a0c14f2bd24b3cb4a8fc7bd0309377.jpg
رابط التقديم لـ رياض الأطفال بكافة المدارس «حكومي ولغات» | المصري اليوم https://mediaaws.almasryalyoum.com/news/large/2023/05/07/2099249_0.jpg
سي سي وما سي سي - أغاني أطفال | قناة وناسة - YouTube https://i.ytimg.com/vi/FrNJnk1q5Fg/maxresdefault.jpg
أطفال سعداء بعد المدرسة, المدرسة, مدرسة, سعيدة PNG صورة للتحميل مجانا https://png.pngtree.com/png-vector/20230423/ourlarge/pngtree-happy-kids-after-school-png-image_6719856.png
أطفال يأكلون نزهة في الغابة, الأطفال في نزهة, صور عالية الدقة, مصنع ... https://png.pngtree.com/background/20230516/original/pngtree-children-eating-a-picnic-in-the-woods-picture-image_2612177.jpg
دروس من صمود الأطفال الفلسطينيين يمكن تعليمها لأبنائنا - اليوم السابع https://img.youm7.com/xlarge/202310210753155315.jpg
مكتبات الأطفال في دبي : أم سقيم - هور العنز - الراشدية - الطوار | ماي بيوت https://mybayutcdn.bayut.co

In [9]:
مثال باستخدام منطقة عربية:
results = ddgs.images(query="أطفال", region="xa-ar", safesearch="on", max_results=10)

In [11]:
from ddgs import DDGS

# إنشاء كائن DDGS
ddgs = DDGS()

# البحث عن صور "أطفال"
results = ddgs.images(
    query="أطفال",
    region="us-en",
    safesearch="off",
    max_results=10
)

# طباعة النتائج
for img in results:
    print(img["title"], img["image"])

إنتاج أغنية بصوت أطفال للحضانة أو للتعليم أو للمسرحيات - خمسات https://khamsat.hsoubcdn.com/images/services/2848438/e9a0c14f2bd24b3cb4a8fc7bd0309377.jpg
رابط التقديم لـ رياض الأطفال بكافة المدارس «حكومي ولغات» | المصري اليوم https://mediaaws.almasryalyoum.com/news/large/2023/05/07/2099249_0.jpg
سي سي وما سي سي - أغاني أطفال | قناة وناسة - YouTube https://i.ytimg.com/vi/FrNJnk1q5Fg/maxresdefault.jpg
أطفال سعداء بعد المدرسة, المدرسة, مدرسة, سعيدة PNG صورة للتحميل مجانا https://png.pngtree.com/png-vector/20230423/ourlarge/pngtree-happy-kids-after-school-png-image_6719856.png
أطفال يأكلون نزهة في الغابة, الأطفال في نزهة, صور عالية الدقة, مصنع ... https://png.pngtree.com/background/20230516/original/pngtree-children-eating-a-picnic-in-the-woods-picture-image_2612177.jpg
دروس من صمود الأطفال الفلسطينيين يمكن تعليمها لأبنائنا - اليوم السابع https://img.youm7.com/xlarge/202310210753155315.jpg
مكتبات الأطفال في دبي : أم سقيم - هور العنز - الراشدية - الطوار | ماي بيوت https://mybayutcdn.bayut.co

In [ ]:
from ddgs import DDGS

# إنشاء كائن DDGS
ddgs = DDGS()

# البحث عن صور "أطفال"
results = ddgs.images(
    query="",
    region="us-en",
    safesearch="off",
    max_results=10
)

# طباعة النتائج
for img in results:
    print(img["title"], img["image"])

In [13]:
from ddgs import DDGS
results = DDGS().news(query="زلزال تركيا", region="xa-ar", max_results=5)
for r in results:
    print(r["title"], r["url"])

زلزال بقوة 3.8 درجة على مقياس ريختر يهز أنطاليا التركية https://www.youm7.com/story/2025/12/16/زلزال-بقوة-3-8-درجة-على-مقياس-ريختر-يهز-أنطاليا/7236335
زلزال تركيا وسوريا: كيف هي أحوال الناجين بعد عام من الكارثة؟ https://www.bbc.com/arabic/articles/cjk6l7m16xlo
تركيا: زلزال بقوة 6,1 درجات يضرب غرب البلاد https://www.france24.com/ar/أوروبا/20250810-تركيا-زلزال-بقوة-6-1-درجات-يضرب-غرب-البلاد-كارثة-طبيعة
"غوغل" تعترف بفشل نظامها في تحذير 10 ملايين من شدة زلزال تركيا المدمر في 2023 https://www.alarabiya.net/technology/in-focus/2025/07/28/غوغل-تعترف-بفشل-نظامها-في-تحذير-10-ملايين-من-شدة-زلزال-تركيا-عام-2023-المميت
زلزال بقوة 4.2 درجة يضرب أنطاكيا جنوب تركيا https://www.msn.com/ar-ae/news/other/زلزال-بقوة-4-2-درجة-يضرب-أنطاكيا-جنوب-تركيا/ar-AA1Sp812?ocid=BingNewsVerp


In [14]:
from ddgs import DDGS
results = DDGS().news(query="تركيا", region="xa-ar", max_results=5)
for r in results:
    print(r["title"], r["url"])

تركيا تتوقع بدء المرحلة الثانية من اتفاق غزة مطلع عام 2026 https://www.msn.com/ar-sa/news/other/تركيا-تتوقع-بدء-المرحلة-الثانية-من-اتفاق-غزة-مطلع-عام-2026/ar-AA1SQO3M?ocid=BingNewsVerp
القبض على ملكة جمال تركيا ومطرب راب شهير في قضية المخدرات: ما القصة؟ https://www.msn.com/ar-sa/entertainment/celebrity-news/القبض-على-ملكة-جمال-تركيا-ومطرب-راب-شهير-في-قضية-المخدرات-ما-القصة/ar-AA1T9DFx?ocid=BingNewsVerp
تركيا: تمويل جديد من روسيا بتسعة مليارات دولار لمحطة آق قويو النووية https://www.msn.com/ar-ae/money/news/تركيا-تمويل-جديد-من-روسيا-بتسعة-مليارات-دولار-لمحطة-آق-قويو-النووية/ar-AA1T41ZH?ocid=BingNewsVerp
تركيا: الإبلاغ عن عطل كهربائي قبيل تحطم طائرة تقل عسكريين ليبيين https://www.msn.com/ar-ae/news/featured/تركيا-الإبلاغ-عن-عطل-كهربائي-قبيل-تحطم-طائرة-تقل-عسكريين-ليبيين/ar-AA1SWxFv?ocid=BingNewsVerp
زوجة أكرم إمام أوغلو: الديموقراطية في تركيا أصيبت بجرح بالغ لكننا لن نفقد الأمل https://www.msn.com/ar-eg/news/world/زوجة-أكرم-إمام-أوغلو-الديموقراطية-في-تركيا-أصيبت-بجرح-بالغ-لكننا-لن-نفقد-ا

In [15]:
results = DDGS().videos(
    query="تعلم البرمجة",
    region="xa-ar",
    safesearch="on",
    duration="medium",
    max_results=5
)
for v in results:
    print(v["title"], v["content"])

تعلم البرمجة من الصفر - الدرس 1 - إعداد واجهة العمل https://www.youtube.com/watch?v=_FeRPlP_WIw
كيف تعلمت البرمجة؟ أول برنامج أعمله بنفسي بالكامل؟ https://www.youtube.com/watch?v=m0jyjCyFw_c
تعلم البرمجة و كيف تبدأ في البرمجة؟ يعني ايه برمجة؟ ابدأ بأي لغة برمجة؟ | How to start programming https://www.youtube.com/watch?v=k7TumDEPBNc
تعلم البرمجة _ كيف أبدأ تعلم البرمجة ؟ من أين أبدأ وماذا أتعلم ؟ وكيف اكسب ؟ (الكورس الثلاثي) https://www.youtube.com/watch?v=YrY2t3ehGmA
ماهو syntax: تعلم البرمجة من الصفر للمبتدئين ٤ https://www.youtube.com/watch?v=NU_IdBZq1qc


In [16]:
results = DDGS().books(query="علم النفس الاجتماعي", max_results=5)
for b in results:
    print(b["title"], b["url"])

علم النفس الاجتماعي [Arabic] https://annas-archive.li/md5/2027c95a607cfb5138da6bff991cb930
علم النفس الاجتماعي المعرفي: مقدمة [Arabic] https://annas-archive.li/md5/ba44abf6908124541b967d5fdc5a7800
مقدّمة إلى علم النفس الاجتماعي وفِكر غوستاف لو بون https://annas-archive.li/md5/29118e2c4e459d156ce6b260e7b318f8
علم النفس الاجتماعي [Arabic] https://annas-archive.li/md5/661526727db5bb1c921d24cd492c9d69
علم النفس الاجتماعي [Arabic] https://annas-archive.li/md5/aa918ed2a26b390052b3fb03463a37a1


In [17]:
# مثال: البحث عن "علم النفس الاجتماعي" داخل موقع goodreads.com
results = DDGS().text(
    query='علم النفس الاجتماعي site:goodreads.com',
    region='xa-ar',
    max_results=5,
    backend='google'  # أو 'bing'
)

for r in results:
    print(r["title"], r["href"])

Popular علم النفس الاجتماعي Books - Goodreads https://www.goodreads.com/shelf/show/علم-النفس-الاجتماعي
علم النفس الاجتماعى by حامد عبد السلام زهران - Goodreads https://www.goodreads.com/book/show/18081104
مقدمة في علم النفس الاجتماعي ودراسات المسلمين by أحمد محمد عامر ... https://www.goodreads.com/book/show/18341646
علم النفس الاجتماعي by خليل عبد الرحمن المعايطة | Goodreads https://www.goodreads.com/book/show/23121709
المصطفى حدية (Author of قضايا في علم النفس الاجتماعي) - Goodreads https://www.goodreads.com/author/show/14067139._


In [18]:
results = DDGS().text(
    query='علم النفس الاجتماعي site:books.google.com',
    region='xa-ar',
    max_results=5
)

In [2]:
from ddgs import DDGS
# مثال: البحث عن "علم النفس الاجتماعي" داخل موقع goodreads.com
results = DDGS().text(
    query='علم النفس الاجتماعي site:goodreads.com',
    region='xa-ar',
    max_results=5,
    backend='google'  # أو 'bing'
)

for r in results:
    print(r["title"], r["href"])

Popular علم النفس الاجتماعي Books - Goodreads https://www.goodreads.com/shelf/show/علم-النفس-الاجتماعي
علم النفس الاجتماعى by حامد عبد السلام زهران - Goodreads https://www.goodreads.com/book/show/18081104
مقدمة في علم النفس الاجتماعي ودراسات المسلمين by أحمد محمد عامر ... https://www.goodreads.com/book/show/18341646
علم النفس الاجتماعي by خليل عبد الرحمن المعايطة | Goodreads https://www.goodreads.com/book/show/23121709
المصطفى حدية (Author of قضايا في علم النفس الاجتماعي) - Goodreads https://www.goodreads.com/author/show/14067139._


In [3]:
results = DDGS().text(
    query='علم النفس الاجتماعي site:books.google.com',
    region='xa-ar',
    max_results=5
)
for r in results:
    print(r["title"], r["href"])

أسس علم النفس الاجتماعي - أحمد محمد الزعبي - Google Books https://books.google.com/books/about/أسس_علم_النفس_الاجتما.html?id=pD8rDAAAQBAJ
علم النفس الإجتماعي - سامي محسن الختاتنة , فاطمة عبد الرحيم ... https://books.google.com/books/about/علم_النفس_الإجتماعي.html?id=dh-sDQAAQBAJ
علم النفس الاجتماعي؛ أصوله ومبادئه - عبد الفتاح محمد دويدار ... علم النفس الاجتماعي: أسسه وتطبيقاته - درويش، زين العابدين ... علم النفس الاجتماعي - احمد عبد اللطيف وحيد - Google Books علم النفس الاجتماعي المعاصر: مدخل معرفي - عبد الرحمن، محمد ... علم النفس الاجتماعي : اسسه وتطبيقاته - زين العابدين درويش ... https://books.google.com/books/about/علم_النفس_الاجتماعي_أ.html?id=sSwlngEACAAJ
علم النفس الاجتماعي: أسسه وتطبيقاته - درويش، زين العابدين ... علم النفس الاجتماعي - احمد عبد اللطيف وحيد - Google Books علم النفس الاجتماعي المعاصر: مدخل معرفي - عبد الرحمن، محمد ... علم النفس الاجتماعي : اسسه وتطبيقاته - زين العابدين درويش ... https://books.google.com/books/about/علم_النفس_الاجتماعي.html?id=IKQ6wgEACAAJ
علم النف

    لا يمكنك تغيير مصدر DDGS().books() — دومًا يُرجع نتائج من Anna’s Archive.
    لاستهداف مواقع أخرى، استخدم DDGS().text() مع site:example.com.
    Anna’s Archive هو أرشيف تجميعي يحتوي على كتب من مصادر متعددة (مثل LibGen، Z-Library، Internet Archive... إلخ)، لذا قد تكون النتائج شاملة رغم كونها من "موقع واحد".

هل تريد مساعدتي في كتابة بحث نصي مستهدف على موقع معين (مثل Amazon أو Goodreads)؟

https://chat.qwen.ai/c/b31b0d35-4b50-4ba5-83a2-a51dadff8102

In [4]:
from ddgs import DDGS

# إنشاء كائن DDGS
ddgs = DDGS()

# البحث النصي عن "علم النفس الاجتماعي"
results = ddgs.text(
    query="علم النفس الاجتماعي",
    region="xa-ar",        # منطقة عربية
    safesearch="on",       # فلترة المحتوى غير اللائق
    timelimit="y",         # اختياري: نتائج من السنة الماضية فقط
    max_results=10,
    backend="google"       # يمكنك تغييره إلى: 'bing', 'duckduckgo', 'brave', 'yandex'
)

# عرض النتائج
for r in results:
    print(f"العنوان: {r['title']}")
    print(f"الرابط: {r['href']}")
    print(f"الوصف: {r.get('body', 'لا يوجد وصف')}\n")

العنوان: علم النفس الاجتماعي وفهم التفاعلات البشرية: منظور تحليلي شامل
الرابط: https://www.mojtamaafikr.com/2025/06/social-psychology-understanding-human-interactions.html
الوصف: Jun 18, 2025 · ج1: علم النفس الاجتماعي هو الدراسة العلمية لكيفية تأثر أفكار ومشاعر وسلوكيات الأفراد بوجود الآخرين الفعلي أو المتخيل أو الضمني. يركز على التفاعل بين الفرد ...

العنوان: Video: علم النفس الاجتماعي والسلوك الفردي - JoVE
الرابط: https://www.jove.com/ar/science-education/v/19864/social-psychology-and-individual-behavior
الوصف: Jul 20, 2025 · يدرس علم النفس الاجتماعي كيف تشكل ديناميكيات المجموعة والعواطف والتأثيرات الثقافية الأفعال الفردية واتخاذ القرار. تتفاعل هذه العناصر لتشكيل أنماط سلوكية ...

العنوان: علم النفس الاجتماعي - وليم لامبرت و وولاس "يسعى هذا الكتاب إلى تقديم ...
الرابط: https://www.facebook.com/AghrabM/posts/علم-النفس-الاجتماعي-وليم-لامبرت-و-وولاسيسعى-هذا-الكتاب-إلى-تقديم-عرض-شامل-لعلم-ا/1204094241732262/
الوصف: Aug 22, 2025 · يوضح المؤلفان أنّ علم النفس الاجتماعي ليس مجرد دراسة للفرد

In [5]:
query = "علم النفس الاجتماعي site:goodreads.com"

In [6]:
query = "مذكرة في علم النفس filetype:pdf"

In [7]:
results = ddgs.text(
    query="علم النفس الاجتماعي site:amazon.com",
    region="us-en",
    backend="google",
    max_results=5
)

البحث فى التكست
11:43 PM

بالطبع! يمكنك استخدام دالة text() من مكتبة ddgs للبحث النصي (مثل بحث Google العادي) في أي موضوع، وتحديد محرك البحث الذي تريده (مثل google، bing، duckduckgo، brave، إلخ).

إليك مثالًا بسيطًا للبحث النصي:

python
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20

ميزات مهمة في text():
1. البحث داخل موقع معيّن (باستخدام site:):

python
1

→ يعرض نتائج فقط من Goodreads.
2. البحث عن ملفات PDF:

python
1

3. تصفية حسب الوقت (timelimit):

    'd': آخر 24 ساعة  
    'w': آخر أسبوع  
    'm': آخر شهر  
    'y': آخر سنة

4. اختيار المحرك (backend):

    'google': نتائج دقيقة وواسعة
    'bing': يدعم العربية جيدًا
    'duckduckgo': خصوصية عالية
    'brave': بديل سريع وآمن

مثال: البحث في Amazon عن كتب:

python
1
2
3
4
5
6

    ملاحظة: لا تعمل جميع المواقع بشكل جيد مع site: (خاصة إذا منعت الزحف)، لكنها فعّالة في معظم الحالات.

هل تريد أن أكتب لك كودًا للبحث في موقع معين (مثل Wikipedia، Google Books، أو Internet Archive)؟
ما المقصود بالبحث في النص؟
كيف يمكنني مساعدتك في البحث عن محتوى معين؟
هل ترغب في تحليل نص معين أو البحث عن كلمات مفتاحية؟
Image Edit
Web Dev
Image Generation


In [8]:
from ddgs import DDGS

ddgs = DDGS()

# البحث في ويكيبيديا العربية فقط
results = ddgs.text(
    query="علم النفس الاجتماعي site:ar.wikipedia.org",
    region="xa-ar",
    backend="google",       # Google يتعامل جيدًا مع "site:"
    max_results=5,
    safesearch="on"
)

for r in results:
    print(f"العنوان: {r['title']}")
    print(f"الرابط: {r['href']}")
    print(f"الوصف: {r.get('body', '—')}\n")

العنوان: علم النفس الاجتماعي - ويكيبيديا
الرابط: https://ar.wikipedia.org/wiki/علم_النفس_الاجتماعي
الوصف: علم النفس الاجتماعي هو فرع من فروع علم النفس يدرس السلوك الاجتماعي للفرد والجماعة، كالاستجابة للمثيرات الاجتماعية، وهدفه بناء مجتمع أفضل قائم على فهم سلوك الفرد ... أنواع الجماعات · الهدف العلمي لعلم النفس الاجتماعي

العنوان: علم النفس الاجتماعي (علم الاجتماع) - ويكيبيديا
الرابط: https://ar.wikipedia.org/wiki/علم_النفس_الاجتماعي_(علم_الاجتماع)
الوصف: ولد علم النفس الاجتماعي السوسيولوجي في عام 1902 بالدراسة الفارقة التي أجراها عالم الاجتماع تشارليز هارتون كولي، بعنوان الطبيعة البشرية والرتبة الاجتماعية، ...

العنوان: علم نفس مجتمعي - ويكيبيديا
الرابط: https://ar.wikipedia.org/wiki/علم_نفس_مجتمعي
الوصف: تُطرَح فكرة علم النفس المجتمعي كعامل توازن أمام تركيز علم النفس الاجتماعي السائد على دراسة أفكار المرء ومشاعره وأفعاله، مع عدم الانتباه كثيرًا لدراسة البيئة ...

العنوان: تصنيف:علم النفس الاجتماعي - ويكيبيديا
الرابط: https://ar.wikipedia.org/wiki/تصنيف:علم_النفس_الاجتماعي
الوصف: ن · ن

In [9]:
results = ddgs.text(
    query="social psychology site:en.wikipedia.org",
    region="us-en",
    backend="google",
    max_results=5
)

for r in results:
    print(f"Title: {r['title']}")
    print(f"URL: {r['href']}")
    print(f"Snippet: {r.get('body', '—')}\n")

Title: Social psychology - Wikipedia
URL: https://en.wikipedia.org/wiki/Social_psychology
Snippet: Social psychology is the methodical study of how thoughts, feelings, and behaviors are influenced by the actual, imagined, or implied presence of others. History · Social cognition · Social influence · Research

Title: List of social psychology theories - Wikipedia
URL: https://en.wikipedia.org/wiki/List_of_social_psychology_theories
Snippet: Social psychology utilizes a wide range of specific theories for various kinds of social and cognitive phenomena. Here is a sampling of some of the more ...

Title: social psychology - Wikidata
URL: https://en.wikipedia.org/wiki/d:Q161272
Snippet: a branch of psychology that combines concepts from psychology and sociology and that focuses on social influence on human behavior.

Title: Societal psychology - Wikipedia
URL: https://en.wikipedia.org/wiki/Societal_psychology
Snippet: Societal psychology is a development within social psychology which emph

💡 ملاحظات:

    استخدام site:ar.wikipedia.org يقيّد النتائج بالموقع الرسمي لويكيبيديا العربية فقط.
    إذا لم تظهر نتائج، جرّب تقليل الكلمات في الاستعلام (مثل: "علم النفس site:ar.wikipedia.org").
    محرك google هو الأفضل مع site:، لكن يمكنك تجربة bing أيضًا.

هل تريد أن أُخرِج لك أول نتيجة كرابط مباشر أو أفتحها تلقائيًّا؟